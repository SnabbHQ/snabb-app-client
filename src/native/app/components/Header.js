/**
* # Header.js
*
 * This component initially displays a image. But when clicked, things
 * get interesting.
 *
 * On the initial display after being clicked, the
 * textinput will display the current ```state``` of the application.
 *
 * The button will be enabled and if clicked, whatever state is now
 * contained in the textinput will be processed and the application
 * will be restored to that state.
 *
 * By pasting in a previous state, the application will reset to that
 * state
 *
 * When the mark image is clicked, it is just toggled to display or hide.
*/
'use strict'

import React, {PropTypes} from "react"
import {ActivityIndicator, Image, StyleSheet, Text, TextInput, TouchableHighlight, View} from "react-native"
import FormButton from "./FormButton"

/**
 * ### propTypes
 * * isFetching: display the spinner if true
 * * showState: should the JSON state, currentState, be displayed
 * * currentState: the JSON state
 * * onGetState: the action to call to get the current state
 * * onSetState: the action to call to set the state
 */
const propTypes = {
  isFetching: PropTypes.bool,
  showState: PropTypes.bool,
  currentState: PropTypes.object,
  onGetState: PropTypes.func,
  onSetState: PropTypes.func
}

class Header extends React.Component {

  constructor(props: P) {
    super(props)

    this.state = {
      text: '',
      isDisabled: true
    }
  }

  /**
   * ### _onPressMark
   * Call the onGetState action passing the state prop
   */
  _onPressMark () {
    this.props.onGetState(!this.props.showState)
  }

  /**
   * ### _onChangeText
   * when the textinput value changes, set the state for that component
   */
  _onChangeText (text) {
    this.setState({
      text,
      isDisabled: false
    })
  }

  /**
   * ### _updateStateButtonPress
   * When the button for the state is pressed, call ```onSetState```
   */
  _updateStateButtonPress () {
    this.props.onSetState(this.state.text)
  }

  /**
   * ### render
   *
   * if showState, stringify the currentState and display it to the
   * browser for copying. Then display to the user.
   *
   * When the value of the input changes, call ```_onChangeText```
   *
   * When the 'Update State' button is pressed, we're off to the
   * races with Hot Loading...just call the
   * ```_updateStateButtonPress``` and away we go...
   *
   */
  render () {
    let displayText
    if (this.props.showState) {
      displayText = JSON.stringify(this.props.currentState)

      console.log(displayText)
    }

    return (
      <View>
        <View style={styles.header}>

          <TouchableHighlight onPress={this._onPressMark}>

            <Image style={styles.mark}
              source={require('./img/snabb.png')}
            />
          </TouchableHighlight>
            {this.props.isFetching
             ? <ActivityIndicator animating size='large' />
             : null
          }

        </View>
            {this.props.showState
             ? <View style={styles.container}>
               <Text>{I18n.t('Header.current_state')} ({I18n.t('Header.see_console')})</Text>
               <TextInput style={{height: 100, borderColor: 'gray', borderWidth: 1}}
                 value={displayText}
                 editable
                 multiline
                 onChangeText={(text) => this._onChangeText(text)}
                 numberOfLines={20} />

               <View style={{
                 marginTop: 10
               }}>
                 <FormButton isDisabled={this.state.isDisabled}
                   onPress={this._updateStateButtonPress}
                   buttonText={I18n.t('Header.update_state')} />
               </View>
             </View>
            : null}
      </View>
    )
  }
}

Header.propTypes = propTypes

/**
 * ## Styles
 */
let styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    flex: 1,
    marginTop: 10
  },
  header: {
    marginTop: 20,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'transparent'
  },
  mark: {
    height: 100,
    width: 100
  }

})


export default Header
