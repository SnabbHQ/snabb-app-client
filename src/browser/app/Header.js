/* @flow */
import type { State, User } from '../../common/types';
import React, { PropTypes } from 'react';
import linksMessages from '../../common/app/linksMessages';
import { NewDeliveryButton } from '../job/components';
import { FormattedMessage } from 'react-intl';
import { Fixed, Link, Text, Box, Image, Space } from '../app/components';
import { connect } from 'react-redux';

// $FlowFixMe
const logo = require('../../../assets/images/logo.svg');

// $FlowFixMe
const clientPhoto = require('../../../assets/images/clientPhotoDefaultSmall.svg');


type HeaderProps = {
  viewer: ?User,
};

type HeaderLinkProps = {
  exactly?: boolean,
  to: string,
  message: Object,
};

const HeaderLink = ({ exactly, to, message }: HeaderLinkProps) => (
  <Link
    antialiasing
    bold
    color="white"
    exactly={exactly}
    marginHorizontal="1.5em"
    to={to}
  >
    <FormattedMessage {...message} />
  </Link>
);

const Header = ({ user }: HeaderProps, { router }: Object) => {
  const onProfileImageClick = () => {
    router.transitionTo('/profile/details');
  };

  return (
    <Fixed top left right zIndex={5}>
      <Box
        backgroundColor="black"
        display="flex"
        flexWrap="wrap"
        paddingVertical={0.3}
        boxShadow="0 1px 2px rgba(0,0,0,0.15)"
      >
        <Space x={1} />
        <Box
          display="flex"
          alignItems="center"
        >
          <Image
            alt="Snabb logo"
            height={50}
            width={50}
            src={logo}
          />
          <Space x={1} />
          <Text color="white" size={1}>Snabb</Text>
        </Box>
        <Space auto />
        <HeaderLink to="/active" message={linksMessages.active} />
        <HeaderLink to="/history" message={linksMessages.history} />
        <Space auto />
        <Box
          display="flex"
          alignItems="center"
        >
          <NewDeliveryButton />
          <Space x={1} />
          <Image
            height={40}
            width={40}
            alt="Snabb"
            src={clientPhoto}
            onClick={onProfileImageClick}
          />
        </Box>
        <Space x={1} />
      </Box>
    </Fixed>
  );
};

Header.contextTypes = {
  router: PropTypes.object.isRequired,
};

export default connect(
  (state: State) => ({
    user: state.user,
  }),
)(Header);

