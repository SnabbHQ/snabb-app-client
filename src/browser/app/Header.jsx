/* @flow */
import type {State, User} from "../../common/types"
import React from "react"
import linksMessages from "../../common/app/linksMessages"
import {NewDeliveryButton} from '../job/components'
import {FormattedMessage} from "react-intl"
import {Link, Box, Image} from "../app/components"
import {Fixed, Space} from "../app/components-old"
import {connect} from "react-redux"

// $FlowFixMe
const logo = require('../../../assets/images/logo.svg')

// $FlowFixMe
const clientPhoto = require('../../../assets/images/clientPhotoDefaultSmall.svg')


type HeaderProps = {
  viewer: ?User,
};

type HeaderLinkProps = {
  exactly?: boolean,
  to: string,
  message: Object,
};

const HeaderLink = ({exactly, to, message}: HeaderLinkProps) => (
  <Link
    bold
    color="white"
    exactly={exactly}
    marginHorizontal="1.5em"
    to={to}
  >
    <FormattedMessage {...message} />
  </Link>
);

const Header = ({viewer}: HeaderProps) => (
  <Fixed top left right zIndex={2}>
    <Box
      backgroundColor="primary"
      display="flex"
      flexWrap="wrap"
      marginBottom={0.5}
      paddingVertical={0.5}
    >
      <Space x={2}/>
      <Box
        display="flex"
        alignItems="center"
      >
        <Image
          alt="Snabb logo"
          height={50}
          width={50}
          src={logo}
        />
        <Space x={2}/>
        <HeaderLink exactly to="/" message={linksMessages.active}/>
        <HeaderLink to="/scheduled" message={linksMessages.scheduled}/>
        <HeaderLink to="/past" message={linksMessages.past}/>
      </Box>
      <Space auto/>
      <Box
        display="flex"
        alignItems="center"
      >
        <NewDeliveryButton/>
        <Space x={2}/>
        <Image
          height={40}
          width={40}
          alt="Snabb"
          src={clientPhoto}/>
      </Box>
      <Space x={2}/>
    </Box>
  </Fixed>
);

export default connect(
  (state: State) => ({
    viewer: {}//state.users.viewer,
  }),
)(Header);


// const HeaderLink = ({ exactly, to, message }: HeaderLinkProps) => (
//   <Link
//     bold
//     color="white"
//     exactly={exactly}
//     marginHorizontal="small"
//     to={to}
//   >
//     <FormattedMessage {...message} />
//   </Link>
// )
//
// class Header extends React.Component {
//
//   state: {
//     dropdownOpen: boolean
//   }
//   styles: Object
//   toggle: () => void
//
//
//   constructor(props) {
//     super(props)
//
//     this.state = {
//       dropdownOpen: false
//     }
//
//     this.styles = {
//       toolbar: {
//         zIndex: 1,
//         justifyContent: 'space-between',
//         flexWrap: 'wrap',
//         boxShadow: '0 2px 6px 0 rgba(0,0,0,.50)',
//         webkitBoxShadow: '0 2px 6px 0 rgba(0,0,0,.50)',
//         mozBoxShadow: '0 2px 6px 0 rgba(0,0,0,.50)',
//       },
//       headerLink: {
//         hover: {borderRadius: 4, },//backgroundColor: rebass.colors.grey},
//         active: {}//color: rebass.colors.accent},
//       }
//     }
//
//     this.toggle = this.toggle.bind(this)
//   }
//
//   toggle(key) {
//     return (e) => {
//       const val = !this.state[key]
//       this.setState({ [key]: val })
//     }
//   }
//
//   render() {
//     return (
//
// <Toolbar style={this.styles.toolbar}>
//   <Flex align="center">
//     <Space x={2}/>
//     <Image
//       alt="Snabb"
//       src={logo}/>
//     <Space x={4}/>
//     <Link p={1} inverted exactly style={this.styles.headerLink} to="/">
//       <FormattedMessage {...linksMessages.active} />
//     </Link>
//     <Space x={2}/>
//     <Link p={1} inverted exactly style={this.styles.headerLink} to="/scheduled">
//       <FormattedMessage {...linksMessages.scheduled} />
//     </Link>
//     <Space x={2}/>
//     <Link p={1} inverted exactly style={this.styles.headerLink} to="/Past">
//       <FormattedMessage {...linksMessages.past} />
//     </Link>
//     <Space x={2}/>
//   </Flex>
//   <Flex>
//     <NewDeliveryButton/>
//     <Space x={2}/>
//     <Dropdown>
//       <Image
//         alt="Snabb"
//         src={clientPhoto}
//         onClick={this.toggle('dropdownOpen')}/>
//       <DropdownMenu
//         onDismiss={this.toggle('dropdownOpen')}
//         open={this.state.dropdownOpen}>
//         <NavItem is="a">
//           Settings
//         </NavItem>
//         <NavItem is="a">
//           Logout
//         </NavItem>
//       </DropdownMenu>
//     </Dropdown>
//   </Flex>
// </Toolbar>
//     )
//   }
//
// }
//
// Header.propTypes = {
//   viewer: React.PropTypes.object,
// }

// export default connect(
//   (state: State) => ({}),
// )(Header)



