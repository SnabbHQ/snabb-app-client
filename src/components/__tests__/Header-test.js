/**
 * # Header-test.js
 *
 * This class tests that the Header component displays correctly
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 *
 */
'use strict'

jest.mock('ActivityIndicator', () => 'ActivityIndicator')

/**
* ## Imports
*/
import 'react-native'
import React from 'react'

import Header from '../Header'

import renderer from 'react/lib/ReactTestRenderer'

/**
 * ## Test
 */
describe('Header', () => {
  /**
   * ### it should be display empty text when not fetching
   * render the header when not fetching
   */
  it('should be display empty text when not fetching', () => {
    const props = {
      isFetching: false
    }
    const tree = renderer.create(<Header {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  /**
   * ### it should be display spinner when fetching
   * When fetching, the GiftedSpinner should display
   */
  it('should be display spinner when fetching', () => {
    const props = {
      isFetching: true
    }

    const tree = renderer.create(<Header {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})// describe Header
