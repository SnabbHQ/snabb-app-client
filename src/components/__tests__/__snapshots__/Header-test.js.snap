exports[`Header should be display empty text when not fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        } />
    </View>
  </View>
</View>
`;

exports[`Header should be display spinner when fetching 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "justifyContent": "center",
        "marginTop": 20,
      }
    }>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
        ]
      }
      testID={undefined}>
      <Image
        source={1}
        style={
          Object {
            "height": 100,
            "width": 100,
          }
        } />
    </View>
    <ActivityIndicator
      animating={true}
      size="large" />
  </View>
</View>
`;
